File: app/src/main/java/mx/edu/upqroo/kristenandroid/viewModels/GradesViewModel.java
package mx.edu.upqroo.kristenandroid.viewModels;

import android.app.Application;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import mx.edu.upqroo.kristenandroid.data.database.entities.Grade;
import mx.edu.upqroo.kristenandroid.data.repositories.GradeRepository;

public class GradesViewModel extends AndroidViewModel {
    private GradeRepository mRepo;

    public GradesViewModel(@NonNull Application application) {
        super(application);
        mRepo = GradeRepository.getInstance(application);
    }

    public LiveData<List<Grade>> getGrades(String userId) {
        return mRepo.getGrades(userId);
    }

    public void updateGradesFromService() {
        mRepo.updateGradesFromService();
    }
}

File: app/src/main/java/mx/edu/upqroo/kristenandroid/viewModels/GradesViewModel.kt
package mx.edu.upqroo.kristenandroid.viewModels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import mx.edu.upqroo.kristenandroid.data.database.entities.Grade
import mx.edu.upqroo.kristenandroid.data.repositories.GradeRepository

class GradesViewModel(application: Application) : AndroidViewModel(application) {
    private val mRepo: GradeRepository = GradeRepository.getInstance(application)

    fun getGrades(userId: String): LiveData<List<Grade>> {
        return mRepo.getGrades(userId)
    }

    fun updateGradesFromService() {
        mRepo.updateGradesFromService()
    }
}
